### PR TITLE
Improve Validation Trigger Equation

### PR DESCRIPTION
**Pull request by nprisbrey**
The original equation to trigger validation was dependent on digits far to the right of the decimal place in floats, which would cause issues when the digits would vary due to computer precision and accuracy. This code was updated to use new variables that have been created since its inception and makes sure we never miss a validation run.

This PR should resolve the problem found in issue #14.

### CODE REVIEW

File: train_model.py [hunk 1]
```diff
@@ -409,12 +409,11 @@ def get_params(self) -> dict:
...
             optimizer.step()
 
-            # Run validation args.validation_freq times per epoch. To do this,
-            # we split up the epoch into arg.validation_freq chunks and run
-            # validation after each chunk is finished.
-            progress_through_chunk = args.val_freq * (batch_idx + 1) \
-                                     / len(train_loader) % 1
-            if progress_through_chunk <= (args.val_freq-1) / len(train_loader):
+            # Run validation args.val_freq times per epoch. To do this, we split
+            # up the epoch into arg.validation_freq chunks and run validation
```
Comment (KimballNJardine) [inline train_model.py]: should be arg.val_freq, not arg.validation_freq

File: train_model.py [hunk 2]
```diff
@@ -409,12 +409,11 @@ def get_params(self) -> dict:
...
+            # after each chunk is finished.
+            if (num_val_runs + 1) / args.val_freq \
```
Comment (KimballNJardine) [inline train_model.py]: If val_freq is set to 0, this will crash.

**Review by KimballNJardine — CHANGES_REQUESTED** [review]
There is a minor issue with the comment block. There is also a divide by 0 issue involving val_freq = 0.

File: train_model.py (hunk 1 above)
Comment (nprisbrey) [reply train_model.py]: Addressed in [80b1bf3](https://github.com/DRAGNLabs/301r_retnet/pull/18/commits/80b1bf3f05f74fc76487b76b804c8cecf6e6f1b9).

File: train_model.py (hunk 2 above)
Comment (nprisbrey) [reply train_model.py]: Good catch! I realize that there were uninitialized variables that gave errors later on if validation was never run as well. I've changed it so that setting `--val-freq=0` is equivalent to saying you don't want any validation at all in [66f000c](https://github.com/DRAGNLabs/301r_retnet/pull/18/commits/66f000c460b548cd3e5a4f7072284d8bce3fc4a3).

**Review by KimballNJardine — APPROVED** [review]
Looks fine to me.
